### PR TITLE
[release/19.x][libc++] Follow-up to "Poison Pills are Too Toxic"

### PR DESCRIPTION
**Pull request by ldionne**
* Update release notes and `Cxx23.html`
* Update `__cpp_lib_ranges` feature test macro

Cherry-pick of #88383 for LLVM 19.

### CODE REVIEW

**Comment by ldionne** [comment]
CC @JMazurkiewicz @frederick-vs-ja 

@frederick-vs-ja Does the FTM change seem correct to you? IIUC this should have been done before the LLVM 19 branch but it slipped through the cracks, so this ought to be correct?

File: libcxx/utils/generate_feature_test_macro_components.py [hunk 1]
```diff
@@ -992,6 +992,7 @@ def add_version_header(tc):
             "name": "__cpp_lib_ranges",
             "values": {
                 "c++20": 202207,
+                "c++23": 202211,  # P2602R2 Poison Pills are Too Toxic
```
Comment (frederick-vs-ja) [inline libcxx/utils/generate_feature_test_macro_components.py:995]: (Not strictly related to this PR) I think the value is fine in C++23 mode, but perhaps it should be `202110L` in C++20 mode as explained in #101715.

Comment (ldionne) [reply libcxx/utils/generate_feature_test_macro_components.py:995]: Ack, we can fix that in a follow-up PR indeed.

File: libcxx/docs/ReleaseNotes/19.rst [hunk 2]
```diff
@@ -77,6 +77,7 @@ Implemented Papers
 - P2602R2 - Poison Pills are Too Toxic
 - P1981R0 - Rename ``leap`` to ``leap_second``
 - P1982R0 - Rename ``link`` to ``time_zone_link``
+- P2602R2 - Poison Pills are Too Toxic (as DR against C++20)
```
Comment (frederick-vs-ja) [inline libcxx/docs/ReleaseNotes/19.rst:80]: The line is duplicated.
```suggestion
- P2602R2 - Poison Pills are Too Toxic (as DR against C++20)
- P1981R0 - Rename ``leap`` to ``leap_second``
- P1982R0 - Rename ``link`` to ``time_zone_link``
```

**Comment by tru** [comment]
Rebasing since we have several CI failures.

**Comment by ldionne** [comment]
I think this is good to go, the CI is as green as it'll get without rebasing onto `release/19.x` again.

**Review by frederick-vs-ja — APPROVED** [review]
Looks like that we should merge this now.

**Pull request merged**